tests/features/joinup_discussion/discussion.subscribe.feature

```diff
@@ -100,3 +102,48 @@ Feature: Subscribing to discussions
+    # relevant fields are changed.
+    Given the mail collector cache is empty
+    And I am logged in as "Dr. Hans Zarkov"
+    When I go to the discussion content "Rare Butter" edit screen
+    And I press "Update"
+    Then 0 e-mails should have been sent
+
+    # When relevant fields of a discussion are changed, the subscribers are
+    # receiving a notifications.
+    Given I go to the discussion content "Rare Butter" edit screen
+    And I fill in "Content" with "The old content was wrong."
+    And I press "Update"
+    And the following email should have been sent:
+      | recipient | dale@example.com                                                                  |
+      | subject   | Joinup: The discussion "Rare Butter" was updated in the space of "Dairy products" |
+      | body      | The discussion "Rare Butter" was updated in the "Dairy products" collection.      |
+    And the following email should have been sent:
+      | recipient | flash@example.com                                                                  |
+      | subject   | Joinup: The discussion "Rare Butter" was updated in the space of "Dairy products" |
+      | body      | The discussion "Rare Butter" was updated in the "Dairy products" collection.      |
+    # The author of the discussion update doesn't receive any notification.
+    But the following email should not have been sent:
+      | recipient | hans@example.com                                                                  |
+      | subject   | Joinup: The discussion "Rare Butter" was updated in the space of "Dairy products" |
+      | body      | The discussion "Rare Butter" was updated in the "Dairy products" collection.      |
+    Then 2 e-mails should have been sent
+
+    # If the discussion is moved from 'validated' to any other state, no
+    # notification will be send, regardless if a relevant field is changed.
+    Given the mail collector cache is empty
+    And I am logged in as a moderator
+    When I go to the discussion content "Rare Butter" edit screen
+    And I fill in "Content" with "Is this change triggering notifications?"
+    And I fill in "Motivation" with "Reporting this content..."
+    And I press "Report"
+    Then the following email should not have been sent:
+      | recipient | dale@example.com                                                                  |
+      | subject   | Joinup: The discussion "Rare Butter" was updated in the space of "Dairy products" |
+      | body      | The discussion "Rare Butter" was updated in the "Dairy products" collection.      |
+    And the following email should not have been sent:
+      | recipient | flash@example.com                                                                  |
+      | subject   | Joinup: The discussion "Rare Butter" was updated in the space of "Dairy products" |
+      | body      | The discussion "Rare Butter" was updated in the "Dairy products" collection.      |
```

Comment:
Let's put another line here "Then 0 e-mails should have been sent" - that makes the test more bullet proof if we change the email template in the future.

Reply:
We cannot. Because other notifications are, beside the scope of this ticket, are sent.